vm/package/deploy.rs

```diff
@@ -84,26 +87,26 @@ impl<'de, N: Network> Deserialize<'de> for DeployRequest<N> {
 }
 
 pub struct DeployResponse<N: Network> {
-    deployment: Deployment<N>,
+    transaction: Transaction<N>,
 }
 
 impl<N: Network> DeployResponse<N> {
     /// Initializes a new deploy response.
-    pub const fn new(deployment: Deployment<N>) -> Self {
-        Self { deployment }
+    pub const fn new(transaction: Transaction<N>) -> Self {
+        Self { transaction }
```

Comment:
Same as above, add a check here that the Transaction is a deployment transaction and not an execution.